### PR TITLE
esp32: Allow boards to configure I2C pins using new macros.

### PR DESCRIPTION
**Pull request by mcauser**
Adds ability to specify default I2C pins in ESP32 board definitions. eg.

/ports/esp32/boards/UM_TINYPICO/mpconfigboard.h
```
#define MICROPY_HW_I2C0_SCL (22)
#define MICROPY_HW_I2C0_SDA (21)
```

The TinyPICO board prefers I2C0 on pins SDA=21, SCL=22, rather than the default SDA=18, SCL=19

This change adds `MICROPY_HW_I2C0_SCL` and `MICROPY_HW_I2C0_SDA` to the UM_TINYPICO board def, used by `machine_i2c.c`, along with UM_FEATHERS2 and UM_TINYS2 boards.

Before:
```
>>> from machine import Pin, I2C
>>> i2c = I2C(0)
>>> i2c
I2C(0, scl=18, sda=19, freq=400000)
>>> i2c1 = I2C(1)
>>> i2c1
I2C(1, scl=25, sda=26, freq=400000)
```

After:
```
>>> from machine import Pin, I2C
>>> i2c = I2C(0)
>>> i2c
I2C(0, scl=22, sda=21, freq=400000)     <-- was 18 / 19
>>> i2c1 = I2C(1)
>>> print(i2c1)
I2C(1, scl=25, sda=26, freq=400000)
```

tinypico-specs-v2-c

**It works!**

To make it consistent with STM32 and SPI board specific pin number overrides, I've rename the I2C defines
|from|to|
|-|-|
|I2C_0_DEFAULT_SCL|MICROPY_HW_I2C0_SCL|
|I2C_0_DEFAULT_SDA|MICROPY_HW_I2C0_SDA|
|I2C_1_DEFAULT_SCL|MICROPY_HW_I2C1_SCL|
|I2C_1_DEFAULT_SDA|MICROPY_HW_I2C1_SDA|

### CODE REVIEW

**Comment by mcauser** [comment]
@UnexpectedMaker Also useful for FeatherS2 + TinyS2 boards

**Comment by UnexpectedMaker** [comment]
Excellent @mcauser ! @dpgeorge this is what we discussed a few meetups ago that SPI was supported but I2C wasn't yet - well Mike added it :)

**Comment by dpgeorge** [comment]
> To make it consistent with SPI board specific pin number overrides, should we rename the I2C defines to match?

Yes, please do this.

Please also separate the commits.  There should be one for each of:
- add ability for a board to configure I2C pins using the new macros
- update the board (UM_TINYPICO) to set the defaults for it

**Comment by mcauser** [comment]
@dpgeorge ready for review.

File: ports/esp32/machine_i2c.c [hunk 1]
```diff
@@ -32,14 +32,28 @@
 
 #include "driver/i2c.h"
 
-#define I2C_0_DEFAULT_SCL (GPIO_NUM_18)
-#define I2C_0_DEFAULT_SDA (GPIO_NUM_19)
+#ifndef MICROPY_HW_I2C0_SCL
+#define MICROPY_HW_I2C0_SCL (GPIO_NUM_18)
+#endif
+
+#ifndef MICROPY_HW_I2C0_SDA
```
Comment (dpgeorge) [inline ports/esp32/machine_i2c.c:39]: Similar to how it's done in `machine_hw_spi.c` (and for stm32 periphs) it's simpler to just require "all or nothing" config: if a board defines `MICROPY_HW_I2C0_SCL` then it must also define `SDA`.  So this #if is not needed.

**Comment by mcauser** [comment]
There's a mixup with the FeatherS2 pins, hold off on merging.

File: ports/esp32/machine_i2c.c (hunk 1 above)
Comment (mcauser) [reply ports/esp32/machine_i2c.c]: @dpgeorge updated to depend on each `CLK` variable like with SPI. 9 lines lighter ⭐

**Comment by dpgeorge** [comment]
Thanks, looks good.  Merged in 538b9a9be5e914eae302d31d8ddedec0f049482a through 605b74f390e1ce9acdbca32d0b3215d37b96852e (I squashed a few related commits)

**Pull request closed without merging**